src/main/java/com/mraof/minestuck/item/crafting/alchemy/GristHelper.java

```diff
@@ -140,14 +155,43 @@ public static void decrease(Level level, PlayerIdentifier player, GristSet set)
 	public static void increase(Level level, PlayerIdentifier player, GristSet set)
 	{
 		Objects.requireNonNull(level);
+		Session session = SessionHandler.get(level).getPlayerSession(player);
+		GristGutter gutter = session.getGristGutter();
+		
 		Objects.requireNonNull(player);
 		Objects.requireNonNull(set);
 		PlayerData data = PlayerSavedData.getData(player, level);
 		NonNegativeGristSet newCache = new NonNegativeGristSet(data.getGristCache());
 		newCache.addGrist(set);
+		int gristCap = rungGrist[rung];
+		
+		GristSet overflowedGrist = limitGristByPlayerRung(level, player, newCache);
+		gutter.addGrist(overflowedGrist);//sends grist overflow to gutter
 		data.setGristCache(newCache);
+		ServerPlayer playerEntity = player.getPlayer(level.getServer());
+		if(playerEntity != null)
+		{
+			gutter.spillGrist(level, playerEntity);//this isn't currently being used
+		}
+	}
+	
+	
+	public static GristSet limitGristByPlayerRung(Level level, PlayerIdentifier player, GristSet set)
+	{
+		int rung = PlayerSavedData.getData(player, level).getEcheladder().getRung();
+		int gristCap = rungGrist[rung];//uses the values in the rungGrist array to determine the current grist cap
+		
+		
+		return set.capGrist(gristCap);//returns the result of capGrist
 	}
 	
+	public static final int[] rungGrist =//never set this below 20.
+			{20,30,40,50,60,70,80,90,100,110,120,130,140,150,160,170,
+					180,190,200,240,250,260,265,270,275,280,285,290,295,300,
+					350,400,450,455,500,1000,2000,3000,4000,5000,6000,7000,
+					8000,9000,10000,20000,30000,40000,50000,500000};
+	// the function that controls how much grist is spliced from the gutter
+	// will crash the game if set below 20
```

Comment:
it would be smart to move all three comments up, so that its clear what it is referring to